src/plugins/data_source_management/public/components/validation/datasource_form_validation.ts

```diff
@@ -62,6 +62,12 @@ export const isTitleValid = (
     isValid.error = i18n.translate('dataSourcesManagement.validation.titleExists', {
       defaultMessage: 'This title is already in use',
     });
+  } else if (title.length > 32) {
```

Comment:
just wonder why put the length check after the already existing check

Reply:
moved to before existence check.